util/src/cli/config.rs

```diff
@@ -1,34 +1,41 @@
+use std::num::NonZeroUsize;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
 use serde::{Deserialize, Serialize};
 
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 #[serde(default)]
 pub struct ThreadPoolConfig {
-    pub rayon_threads: usize,
-    pub tokio_workers: usize,
+    rayon_threads: NonZeroUsize,
+    tokio_workers: NonZeroUsize,
 }
 
 impl ThreadPoolConfig {
     pub fn init_global_rayon_pool(&self) -> Result<(), rayon::ThreadPoolBuildError> {
         rayon::ThreadPoolBuilder::new()
             .stack_size(8 * 1024 * 1024)
-            .thread_name(|_| "rayon_worker".to_string())
-            .num_threads(self.rayon_threads)
+            .thread_name(|id| format!("rayon-global-{id}"))
+            .num_threads(self.rayon_threads.get())
             .build_global()
     }
 
     pub fn build_tokio_runtime(&self) -> std::io::Result<tokio::runtime::Runtime> {
         tokio::runtime::Builder::new_multi_thread()
             .enable_all()
-            .worker_threads(self.tokio_workers)
+            .worker_threads(self.tokio_workers.get())
+            .thread_name_fn(|| {
+                static ATOMIC_ID: AtomicUsize = AtomicUsize::new(0);
+                let id = ATOMIC_ID.fetch_add(1, Ordering::Relaxed);
+                format!("tokio-{id}")
```

Comment:
naming threads will give you ugly flamegraphs, with each worker having a separate stacktrace. 

Better to name main_rayon, mempool_rayon, tokio_worker without indexes.

Eg now it's 1 stack
<img width="1666" height="1324" alt="image" src="https://github.com/user-attachments/assets/0c1a18de-f930-4d4c-92cf-be896988aafb" />
and with your changes it will be 8 umerged stacks for each worker